### PR TITLE
[bn] Launch Bengali localization

### PR DESCRIPTION
**Pull request by mitul3737**
This PR aims to launch the Bengali localization of the Kubernetes website by merging the dev-1.24-bn.1 branch into the main branch.

We have completed all the necessary content and made the required adjustments to launch the Kubernetes website in Bangla.

- https://github.com/kubernetes/website/issues/44993

We also have a perfect deploy preview  :

- https://deploy-preview-45009--kubernetes-io-main-staging.netlify.app/bn/

dev-1.24-bn.1 has been fully reviewed by : 

@asem-hamid 
@mitul3737 
@sajibAdhi 

We kindly request review and approval from 

- sig-docs-localization-reviewers

/cc @a-mccarthy 
/cc @seokho-son 

- sig-docs-leads 

/cc  @divya-mohan0209  
/cc @natalisucks 
/cc @reylejano 
~/cc @sftim~

Thank you for your cooperation and support!

### CODE REVIEW

**Comment by sftim** [comment]
/hold

Localization launch needs approval from website admins

**Review by itsPronay — CHANGES_REQUESTED** [review]
Looks like the change (translation in Bengali) was made by using a translator. Even as a native Bengali speaker, I don't understand most of the terms. Please change 'em

**Comment by asem-hamid** [comment]
@itsPronay  Most of the files are outdated. My team and I are starting fresh. We're working on updating those files and localizing others . Feel free to reach out to  [me](https://kubernetes.slack.com/team/U06DG1EBR9S)  if you notice anything to **Change/Update/Localize** or if you find any **Mistakes** .  I'll take the necessary steps to fix them. .

**Comment by sftim** [comment]
@itsPronay as a courtesy, we avoid reviewing work marked as draft; there are exceptions and I do sometimes offer early feedback.

You are welcome to file new _issues_ to record work that needs doing ahead of launching Bengali, including any recommendation to rework something already localized.

**Comment by sajibAdhi** [comment]
There is a Header Rule Error because the glossary term_id sidecar-container is missing.
 
Screenshot 2024-03-26 223316

**Comment by asem-hamid** [comment]
@sajibAdhi doesn't you have fix that already ?

File: content/bn/releases/patch-releases.md
Comment (dipesh-rawat) [inline content/bn/releases/patch-releases.md:79]: Based on the current failed build logs [here](https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/6637791355f44f00084d1f48)

> 1:19:31 PM: Error: error building site: assemble: "/opt/build/repo/content/bn/releases/patch-releases.md:79:1": failed to extract shortcode: template for shortcode "upcoming-releases" not found

It looks like this `upcoming-releases` shortcode's html file is missing for the branch being built (dev-1.24-bn.1). To resolve this issue, I suggest adding the [upcoming-releases.html (source here)](https://github.com/kubernetes/website/blob/main/layouts/shortcodes/upcoming-releases.html) file from the **main** branch to this pull request branch [dev-1.24-bn.1 (location here)](https://github.com/kubernetes/website/tree/dev-1.24-bn.1/layouts/shortcodes) . Once you do that, the current build error should get resolved.

**Comment by asem-hamid** [comment]
@dipesh-rawat Is there a way to view all errors at once? I believe there may still be more errors there.

**Comment by dipesh-rawat** [comment]
>  Is there a way to view all errors at once? I believe there may still be more errors there.

Typically, the build preview [deploy logs (here)](https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/66385aa898b79f00084711b5) should display all errors encountered. Right now, it's only showing the missing `upcoming-releases` shortcode file error. It's important to note that with each new commit added to the PR branch, the build preview reruns, which might uncover additional failures. Predicting these errors in advance can be challenging, but the logs should capture them all.

**Comment by mitul3737** [comment]
This PR aims to launch the Bengali localization of the Kubernetes website by merging the dev-1.24-bn.1 branch into the main branch.

We have completed all the necessary content and made the required adjustments to launch the Kubernetes website in Bangla.

- https://github.com/kubernetes/website/issues/44993

We also have a perfect deploy preview  :

- https://deploy-preview-45009--kubernetes-io-main-staging.netlify.app/bn/

dev-1.24-bn.1 has been fully reviewed by : 

/cc @asem-hamid 
/cc @mitul3737 
/cc @sajibAdhi 

We kindly request review and approval from 

- sig-docs-localization-reviewers

/cc @a-mccarthy 
/cc @seokho-son 

- sig-docs-leads 

/cc  @divya-mohan0209  
/cc @natalisucks 
/cc @reylejano 
/cc @sftim

Thank you for your cooperation and support!

**Comment by seokho-son** [comment]
I've reviewed all content outside `content/bn` such as hugo.toml, i18n, static, OWNERS_ALIASES, and README.md, and everything looks good to me. +Netlify preview works nicely.

I believe `content/bn` meets the minimal requirements, and individual contents inside `content/bn` have already been reviewed by Bn reviewers:
    - asem-hamid
    - Imtiaz1234
    - mitul3737
    - rajibmitra

Therefore, the content inside this PR, LGTM!

---

However, I am concerned about the commit history. The `903 commits` in the upstream `dev-1.24-bn.1` are not following the SIG-Docs commit squashing policy. `dev-1.24-bn.1` contains too many redundant or irrelevant commits.
My suggestion is to open another PR with one squashed commit for this first release of the Bengali localization.
- For instance, see #44980

Regarding my suggestion for commit squashing, I hope to follow a decision from SIG-Docs leads, and emeritus leads. I'd like to call @kubernetes/sig-docs-leads here to finalize the first live of Bengali localization. :)

File: static/_redirects
Comment (reylejano) [inline static/_redirects:9]:
```suggestion
/bn/docs/      /bn/docs/home/ 301!
```

**Comment by reylejano** [comment]
I also reviewed files outside `content/bn`: OWNERS_ALIASES, README.md, hugo.toml, _redirects. I have one minor [suggestion](https://github.com/kubernetes/website/pull/45009/files#r1595758730)

File: static/_redirects
Comment (asem-hamid) [reply static/_redirects:9]: Done in https://github.com/kubernetes/website/pull/46297

**Comment by reylejano** [comment]
Thank you for addressing my comment
/lgtm 
/assign @reylejano @natalisucks @divya-mohan0209 @seokho-son @a-mccarthy

**Comment by divya-mohan0209** [comment]
Thank you for all your hard work on this! 
/approve

**Comment by divya-mohan0209** [comment]
@seokho-son : Regarding squashing of commits, please could you advise on the precedent followed previously for other localisation go-lives that you have been involved in?

**Comment by seokho-son** [comment]
Hi @divya-mohan0209,
I offered some advice but was not directly involved with the release of previous languages, except for Korean.
I just reviewed how we've managed similar cases in the past.

- https://github.com/kubernetes/website/pull/12998 (67 commits for 270 files)
- https://github.com/kubernetes/website/pull/18569 (3 commits)

(I assume the following languages were based on a previous localization guide)
- https://github.com/kubernetes/website/pull/16404 (35 commits)
- https://github.com/kubernetes/website/pull/13543 (34 commits)
- https://github.com/kubernetes/website/pull/13822 (12 commits)
- https://github.com/kubernetes/website/pull/18419 (5 commits)
- https://github.com/kubernetes/website/pull/10471 (1 commit)

However, my concern is not the number of commits but rather the presence of too many redundant or irrelevant commits in dev-1.24-bn.1, possibly included by the GitHub review suggestion UI as well.

It seems that many localization teams, including the Korean team, have been making efforts to manage commits effectively.
- https://github.com/kubernetes/website/pull/18226 (1 commit)
- https://github.com/kubernetes/website/pull/29998

**Comment by divya-mohan0209** [comment]
I'm okay with it if there have been previous precedents of it working successfully. Again, this is an improvement in how we localise docs that will need general consensus and must be documented. Given the time constraints around the PR, I suggest the following.

- I'd like the other co-chairs, leads, and emeritus leads to chime in on this improvement as the bare minimum. I'm tagging @reylejano @natalisucks @sftim @a-mccarthy @dipesh-rawat here since they have previously been involved/tagged here for reviews.
- By EOD, we will proceed with the recommended approach if a minimum quorum of  50% of the folks tagged in point number 1 assent to this improvement without any technical/procedural concerns.
- If it succeeds without glitches, we will showcase this at our next localisation meeting and PR it as a recommendation in our docs.

**Comment by sftim** [comment]
Although we'd like to see commits done well (it makes the repo easier to manage), if they haven't been then we mostly need to accept that.

I am happy to see Bengali launch if the only qualm about launching it is the commit history on that branch and not follow our norms (not rules, norms) around squashing.

**Comment by sftim** [comment]
As an aside, watch out for commits that touch other parts of the tree and are then reverted within the same PR. That kind of commit makes merges and rebases a pain for everyone else.

(I'm not seeing that here)

**Comment by natalisucks** [comment]
I'm also happy to see the Bengali localization launch and for us to document our desired approach for teams around commits more broadly before stricter reinforcement.

Thanks to our various leads for their reviews here! ❤️

**Comment by seokho-son** [comment]
Ok! I think we've reached a consensus. Let me release the hold and let's look forward to the new language for k8s.io!

/lgtm
/hold cancel

**Comment by asem-hamid** [comment]
Thank you so much for your review, approval, and effort, which allowed us to launch it so quickly.

@seokho-son , @reylejano , @divya-mohan0209 , @sftim , @natalisucks

**Comment by asem-hamid** [comment]
I also want to thank @seokho-son  and @sftim  for all the help and guidance from the beginning until now during the localization progress.

**Pull request merged**
